File: server/Rekindle.Core/Models/ReadingProgress.cs
namespace Rekindle.Core.Models;

public class ReadingProgress
{
    public string UserId { get; set; } = string.Empty;
    public string MediaId { get; set; } = string.Empty;
    public int CurrentPage { get; set; } = 0;
    public bool IsCompleted { get; set; } = false;
    public DateTime LastReadAt { get; set; } = DateTime.UtcNow;
}

File: desktop/lib/core/models/reading_progress.dart
class ReadingProgress {
  final String userId;
  final String mediaId;
  final int currentPage;
  final bool isCompleted;

  const ReadingProgress({
    required this.userId,
    required this.mediaId,
    required this.currentPage,
    required this.isCompleted,
  });

  factory ReadingProgress.fromJson(Map<String, dynamic> j) => ReadingProgress(
        userId: (j['userId'] ?? '') as String,
        mediaId: (j['mediaId'] ?? '') as String,
        currentPage: (j['currentPage'] ?? 0) as int,
        isCompleted: (j['isCompleted'] ?? false) as bool,
      );
}
